### PR TITLE
2.20 prep

### PR DESCRIPTION
**Pull request by edmundnoble**
Includes first changelog draft and cabal package version increment.

### CODE REVIEW

**Review by chessai — APPROVED** [review]

**Comment by chessai** [comment]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 # `chainweb-node` Changelog
 
+## UNRELEASED
+
+Changes:
+
+* A new chainwebVersion called fast-development, which is intended to make the devnet experience faster to get started with for testing transactions. See #1627 for more details.
```
Comment (larskuhtz) [inline CHANGELOG.md:7]: is this correct grammar:

> ... with for ...

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+* Updated from GHC 8.10.7 to GHC 9.6.2. (#1565)
```
Comment (larskuhtz) [inline CHANGELOG.md:8]: Not sure if this is relevant for users? Most users probably don't know what "GHC" is.

Drop this or maybe move it down and/or add a bit more context?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+* Updated to Pact 4.8. (numerous)
```
Comment (larskuhtz) [inline CHANGELOG.md:9]: Add link to pact CHANGELOG?

File: node/ChainwebNode.hs [hunk 4]
```diff
@@ -529,7 +529,7 @@ pkgInfoScopes =
 -- SERVICE DATE for version 2.19
 --
 serviceDate :: Maybe String
-serviceDate = Just "2023-09-07T00:00:00Z"
+serviceDate = Just "2023-11-07T00:00:00Z"
```
Comment (larskuhtz) [inline node/ChainwebNode.hs:532]: These are different days of the week. IIRC we always used to do Thursday, 00:00:00 UTC. Do we want to break with with this tradition?

File: src/Chainweb/Version/Mainnet.hs [hunk 5]
```diff
@@ -139,7 +139,7 @@ mainnet = ChainwebVersion
         Chainweb217Pact -> AllChains (ForkAtBlockHeight $ BlockHeight 3_250_348) -- 2022-12-02T00:00:00+00:00
         Chainweb218Pact -> AllChains (ForkAtBlockHeight $ BlockHeight 3_512_363) -- 2023-03-03 00:00:00+00:00
         Chainweb219Pact -> AllChains (ForkAtBlockHeight $ BlockHeight 3_774_423) -- 2023-06-02 00:00:00+00:00
-        Chainweb220Pact -> AllChains ForkNever
+        Chainweb220Pact -> AllChains (ForkAtBlockHeight $ BlockHeight 4_056_980) -- 2023-09-08 00:00:00+00:00
```
Comment (larskuhtz) [inline src/Chainweb/Version/Mainnet.hs:142]: |    | network   | time                      |   height |   hours after service date |
|---:|:----------|:--------------------------|---------:|---------------------------:|
|  1 | mainnet   | 2023-09-08 00:00:00+00:00 |  4056494 |                         24 |

There has been a drift of 500 blocks (~4h), which seems a bit large. Do we know the reason for this?

File: src/Chainweb/Version/Testnet.hs [hunk 6]
```diff
@@ -119,7 +119,7 @@ testnet = ChainwebVersion
         Chainweb217Pact -> AllChains $ ForkAtBlockHeight $ BlockHeight 2_777_367  -- 2022-12-01 12:00:00+00:00
         Chainweb218Pact -> AllChains $ ForkAtBlockHeight $ BlockHeight 3_038_343  -- 2023-03-02 12:00:00+00:00
         Chainweb219Pact -> AllChains $ ForkAtBlockHeight $ BlockHeight 3_299_753 -- 2023-06-01 12:00:00+00:00
-        Chainweb220Pact -> AllChains ForkNever
+        Chainweb220Pact -> AllChains $ ForkAtBlockHeight $ BlockHeight 3_581_450 -- 2023-09-08 12:00:00+00:00
```
Comment (larskuhtz) [inline src/Chainweb/Version/Testnet.hs:122]: |    | network   | time                      |   height |   hours after service date |
|---:|:----------|:--------------------------|---------:|---------------------------:|
|  0 | testnet   | 2023-09-07 12:00:00+00:00 |  3580966 |                         12 |

There is also a drift of about 450 blocks (~3.5h). Do we know why?

File: tools/calculate-release/CalculateRelease.hs [hunk 7]
```diff
@@ -48,8 +48,8 @@ main = do
 
     putStrLn $ "Mainnet fork height (at " <> show mainnetForkTime <> "): " <> show mainnetForkHeight
     putStrLn $ "Testnet fork height (at " <> show testnetForkTime <> "): " <> show testnetForkHeight
-    let nextServiceDateDay = addGregorianMonthsClip 4 serviceDateDay
-    putStrLn $ "Next service date (+4 months): " <> show nextServiceDateDay
+    let nextServiceDateDay = addGregorianMonthsClip 2 serviceDateDay
+    putStrLn $ "Next service date (+2 months): " <> show nextServiceDateDay
```
Comment (larskuhtz) [inline tools/calculate-release/CalculateRelease.hs:52]: We used to do service releases on Thursdays 00:00:00 UTC. Doing it in terms of month (and not weeks) could result in releases being on weekends or during holidays. Is this intended?

**Review by larskuhtz — COMMENTED** [review]
Left some comments

File: tools/calculate-release/CalculateRelease.hs (hunk 7 above)
Comment (jwiegley) [reply tools/calculate-release/CalculateRelease.hs:52]: We definitely want to keep to Thursdays at 0 UTC, for sanity's sake.

File: CHANGELOG.md (hunk 1 above)
Comment (edmundnoble) [reply CHANGELOG.md:7]: I don't know, but it sounds bad.

File: CHANGELOG.md (hunk 2 above)
Comment (edmundnoble) [reply CHANGELOG.md:8]: I can make an "internal changes" section

File: CHANGELOG.md (hunk 3 above)
Comment (edmundnoble) [reply CHANGELOG.md:9]: Good idea

File: node/ChainwebNode.hs (hunk 4 above)
Comment (edmundnoble) [reply node/ChainwebNode.hs:532]: Oof okay the release calculator will need a fix here.

File: src/Chainweb/Version/Mainnet.hs (hunk 5 above)
Comment (edmundnoble) [reply src/Chainweb/Version/Mainnet.hs:142]: No, this is probably something the release calculator did.

Comment (edmundnoble) [reply src/Chainweb/Version/Mainnet.hs:142]: I don't actually see why the drift has been so big. Looks like we just got unlucky; the release calculator agrees with the height you just posted now.

**Review by larskuhtz — APPROVED** [review]
Approved with one comment

**Pull request merged**